### PR TITLE
zh-CN: Translate some other contents and correct some punctuation usages

### PR DESCRIPTION
**Pull request by TheBlueAlgae**
This commit contains two sections:

1. Translate some contents into Chinese (Simplified).
2. Correct portions of the usages of punctuations.

### CODE REVIEW

File: po/zh-CN.po [hunk 1]
```diff
@@ -4101,118 +3917,108 @@ msgstr "用一个值（例如 `break 8`）来中断 `loop` 并将其输出。"
...
 msgid ""
-"Note that `loop` is the only looping construct which returns a non-trivial "
-"value. This is because it's guaranteed to be entered at least once (unlike "
-"`while` and `for` loops)."
+"Note that `loop` is the only looping construct which returns a non-trivial value. This "
+"is because it's guaranteed to be entered at least once (unlike `while` and `for` loops)."
 msgstr ""
-"请注意，`loop` 是唯一返回有意义的值的循环结构。 这是因为它保证至少被输入一次"
-"（与 `while` 和 `for` 循环不同）。"
+"请注意，`loop` 是唯一返回有意义的值的循环结构。 这是因为它保证至少被输入一次（与 `while` "
```
Comment (emmali01) [inline po/zh-CN.po:3923]: extra space
```suggestion
"请注意，`loop` 是唯一返回有意义的值的循环结构。这是因为它保证至少被输入一次（与 `while` "
```

File: po/zh-CN.po [hunk 2]
```diff
@@ -4101,118 +3917,108 @@ msgstr "用一个值（例如 `break 8`）来中断 `loop` 并将其输出。"
...
 #: src/basic-syntax/static-and-const.md:32
 msgid ""
-"Only functions marked `const` can be called at compile time to generate "
-"`const` values. `const` functions can however be called at runtime."
+"Only functions marked `const` can be called at compile time to generate `const` values. "
+"`const` functions can however be called at runtime."
 msgstr ""
-"在编译时只能调用标记为“const”的函数以生成“const”值。不过，可在运行时调"
-"用“const”函数。"
+"在编译时只能调用标记为“const”的函数以生成“const”值。不过，可在运行时调用“const”函数。"
```
Comment (emmali01) [inline po/zh-CN.po:4012]: I noticed sometimes \`\` is changed to "", while sometimes we keep using \`\` in translated version. Is there a guideline for it?

File: po/zh-CN.po [hunk 3]
```diff
@@ -5538,72 +5299,170 @@ msgid ""
...
 
 #: src/welcome-day-2.md:12
 msgid ""
-"The Standard Library: `String`, `Option` and `Result`, `Vec`, `HashMap`, "
-"`Rc` and `Arc`."
+"The Standard Library: `String`, `Option` and `Result`, `Vec`, `HashMap`, `Rc` and `Arc`."
 msgstr ""
-"标准库: `字符串（String）`, `选项（Option）` 和 `结果（Result）`, `动态数组"
-"（Vec）`, `散列表（HashMap）`, `引用计数（Rc）` 和 `共享引用计数（Arc）`。"
+"标准库：`String`、`Option` 和  `Result`、`动态数组（Vec）`, `散列表（HashMap）`、`引用计"
```
Comment (emmali01) [inline po/zh-CN.po:5460]: use 、 instead of ,？

File: po/zh-CN.po [hunk 4]
```diff
@@ -5678,17 +5535,16 @@ msgid "No guarantee of memory locality."
...
 #: src/memory-management/stack.md:3
-#, fuzzy
 msgid ""
-"Creating a `String` puts fixed-sized metadata on the stack and dynamically "
-"sized data, the actual string, on the heap:"
+"Creating a `String` puts fixed-sized metadata on the stack and dynamically sized data, "
+"the actual string, on the heap:"
 msgstr ""
-"创建 `String` 时将固定大小的数据存储在栈上， 并将动态大小的数据存储在堆上："
+"创建 `String` 时将固定大小的元数据存储在栈上， 并将动态大小的数据（也就是真正的字符串）存"
```
Comment (emmali01) [inline po/zh-CN.po:5546]: extra space

**Comment by mgeisler** [comment]
Hey @TheBlueAlgae, you mentioned thousands of changes and I checked with

```
gh pr diff 1379 | bat -l patch
```

Many of the changes are because you need to run `dprint fmt` after saving the file. That's also why the [format check failed](https://github.com/google/comprehensive-rust/actions/runs/6557133693/job/17808200363?pr=1379) on the PR.

I recently updated [TRANSLATIONS.md](https://github.com/google/comprehensive-rust/blob/main/TRANSLATIONS.md) with more setup instructions — please let me know if you find anything missing there.

**Comment by TheBlueAlgae** [comment]
> Hey @TheBlueAlgae, you mentioned thousands of changes and I checked with
> 
> ```
> gh pr diff 1379 | bat -l patch
> ```
> 
> Many of the changes are because you need to run `dprint fmt` after saving the file. That's also why the [format check failed](https://github.com/google/comprehensive-rust/actions/runs/6557133693/job/17808200363?pr=1379) on the PR.
> 
> I recently updated [TRANSLATIONS.md](https://github.com/google/comprehensive-rust/blob/main/TRANSLATIONS.md) with more setup instructions — please let me know if you find anything missing there.

Hey @mgeisler , I attempted running `dprint fmt zh-CN.po` on my computer but it prompted that "No files found to format with the specified plugins at...". Also, when I initialize `dprint.json`, I do not find a plugin associated with "po" file.

I am using Windows, so what should I do?

File: po/zh-CN.po (hunk 2 above)
Comment (TheBlueAlgae) [reply po/zh-CN.po:4012]: I do not exactly know because this content is not translated by me.😂 I will review them in the future.

**Comment by mgeisler** [comment]
> > Hey @TheBlueAlgae, you mentioned thousands of changes and I checked with
> > ```
> > gh pr diff 1379 | bat -l patch
> > ```
> > 
> > 
> > Many of the changes are because you need to run `dprint fmt` after saving the file. That's also why the [format check failed](https://github.com/google/comprehensive-rust/actions/runs/6557133693/job/17808200363?pr=1379) on the PR.
> > I recently updated [TRANSLATIONS.md](https://github.com/google/comprehensive-rust/blob/main/TRANSLATIONS.md) with more setup instructions — please let me know if you find anything missing there.
> 
> Hey @mgeisler , I attempted running `dprint fmt zh-CN.po` on my computer but it prompted that "No files found to format with the specified plugins at...". Also, when I initialize `dprint.json`, I do not find a plugin associated with "po" file.

There is a `dprint.json` file already in the root of the checkout — you should not initialize a new one.

Also, you can run `dprint fmt` to format everything, or `dprint fmt po/zh-CN.po` to format just your PO file. See [Formatting a Translation](https://github.com/google/comprehensive-rust/blob/main/TRANSLATIONS.md#formatting-a-translation).

> I am using Windows, so what should I do?

I believe all the tools would work on Windows too. You need to install [Gettext](https://mlocati.github.io/articles/gettext-iconv-windows.html) and then you need to install [`dprint`](https://dprint.dev/install/).

**Pull request closed without merging**